mailpoet/lib/Form/DisplayFormInWPContent.php

```diff
@@ -104,11 +106,28 @@ private function shouldDisplay(): bool {
     }
     // this code ensures that we display the form only on a page which is related to single post
     if (!$this->wp->isSingle() && !$this->wp->isPage()) $result = $this->wp->applyFilters('mailpoet_display_form_is_single', false);
+
+    if ($this->displayFormInListingPage()) $result = true;
+
     $noFormsCache = $this->wp->getTransient(DisplayFormInWPContent::NO_FORM_TRANSIENT_KEY);
     if ($noFormsCache === '1') $result = false;
     return $result;
   }
 
+  private function displayFormInListingPage(): bool {
```

Comment:
The code is very specific for Woo products page, but the name of the method is quite generic. I think it would be nicer to name it `displayFormInProductListPage` or somewhat similar. What do you think?

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/4459/commits/d51ae1859d890471ba4efd3529380d10fa45ce37